components/raftstore/src/store/peer.rs

```diff
@@ -2316,6 +2316,11 @@ where
                         let _ = self.get_store().clear_data();
                         self.delay_clean_data = false;
                     }
+                    // only evict when region is initialized
+                    if !self.region().get_peers().is_empty() {
+                        let range = CacheRange::from_region(self.region());
+                        ctx.engines.kv.evict_range(&range);
```

Comment:
This part of code is executed in the critical raft-store thread, the `evict_range` is not a fast enough function, especially when there are many ranges are cached. Here we probably just register the `eviction` is better.